localstack/testing/aws/util.py

```diff
@@ -180,12 +181,17 @@ def base_aws_client_factory(session: boto3.Session) -> ClientFactory:
     config = None
     if os.environ.get("TEST_DISABLE_RETRIES_AND_TIMEOUTS"):
         config = botocore.config.Config(
-            connect_timeout=1_000, read_timeout=1_000, retries={"total_max_attempts": 1}
+            connect_timeout=1_000,
+            read_timeout=1_000,
+            retries={"total_max_attempts": 1},
         )
 
     if os.environ.get("TEST_TARGET") == "AWS_CLOUD":
         return ExternalAwsClientFactory(session=session, config=config)
     else:
+        if config:
+            # Prevent this fixture from using the region configured in system config
+            config = config.merge(botocore.config.Config(region_name=TEST_AWS_REGION_NAME))
         return ExternalClientFactory(session=session, config=config)
```

Comment:
otherwise this would only be set if TEST_DISABLE_RETRIES_AND_TIMEOUTS is set

```suggestion
        if not config:
            config = botocore.config.Config()
            
        # Prevent this fixture from using the region configured in system config
        config = config.merge(botocore.config.Config(region_name=TEST_AWS_REGION_NAME))
        return ExternalClientFactory(session=session, config=config)
```